driver/src/main/native/com/code_intelligence/jazzer/jazzer_preload.c

```diff
@@ -0,0 +1,238 @@
+  DEFINE_LIBC_HOOK(name, ret, params, args)                                \
+                                                                           \
+  typedef ret (*name##_t)(UNWRAP_VA_ARGS params);                          \
+  static _Atomic name##_t name##_real;                                     \
+                                                                           \
+  __attribute__((visibility("default"))) ret name(UNWRAP_VA_ARGS params) { \
+    name##_t name##_real_local =                                           \
+        atomic_load_explicit(&name##_real, memory_order_relaxed);          \
+    if (UNLIKELY(name##_real_local == NULL)) {                             \
+      name##_real_local = dlsym(RTLD_NEXT, #name);                         \
+      atomic_store_explicit(&name##_real, name##_real_local,               \
+                            memory_order_relaxed);                         \
+    }                                                                      \
+    ret result = name##_real_local(UNWRAP_VA_ARGS args);                   \
+    __sanitizer_weak_hook_##name(GET_CALLER_PC(), UNWRAP_VA_ARGS args,     \
+                                 result);                                  \
+    return result;                                                         \
+  }
+
+#elif __APPLE__
+// macOS namespace concept makes it impossible to override symbols in shared
+// library dependencies simply by defining them. Instead, the dynamic linker's
+// interpose feature is used to request that one function, identified by its
+// address, is replaced by another at runtime.
+
+typedef struct {
+  const uintptr_t interceptor;
+  const uintptr_t func;
+} interpose_t;
+
+#define INTERPOSE(_interceptor, _func)                        \
+  __attribute__((used)) static interpose_t _interpose_##_func \
+      __attribute__((section("__DATA,__interpose"))) = {      \
+          (uintptr_t)&_interceptor, (uintptr_t)&_func};
+
+#define DEFINE_LIBC_INTERCEPTOR(name, ret, params, args)               \
+  DEFINE_LIBC_HOOK(name, ret, params, args)                            \
+                                                                       \
+  __attribute__((visibility("default")))                               \
+  ret interposed_##name(UNWRAP_VA_ARGS params) {                       \
+    ret result = name(UNWRAP_VA_ARGS args);                            \
+    __sanitizer_weak_hook_##name(GET_CALLER_PC(), UNWRAP_VA_ARGS args, \
+                                 result);                              \
+    return result;                                                     \
+  }                                                                    \
+                                                                       \
+  INTERPOSE(interposed_##name, name)
+#else
+#error "jazzer_preload is not supported on this OS"
+#endif
+
+DEFINE_LIBC_INTERCEPTOR(bcmp, int, (const void *s1, const void *s2, size_t n),
```

Comment:
Could we get this to work for Windows using [detours](https://github.com/microsoft/Detours)?